toolchain/check/operator.cpp

```diff
@@ -0,0 +1,145 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "toolchain/check/operator.h"
+
+#include "toolchain/check/call.h"
+#include "toolchain/check/context.h"
+#include "toolchain/check/member_access.h"
+#include "toolchain/sem_ir/ids.h"
+#include "toolchain/sem_ir/typed_insts.h"
+
+namespace Carbon::Check {
+
+// Returns the scope of the Carbon package, or Invalid if it's not found.
+//
+// TODO: Consider tracking the Carbon package in SemIR so we don't need to use
+// name lookup to find it.
+static auto GetCarbonPackage(Context& context, Parse::AnyExprId node_id)
+    -> SemIR::NameScopeId {
+  // TODO: If the current package is the `Carbon` package, return
+  // `SemIR::InstId::Package`.
+
+  auto ident_id = context.identifiers().Lookup("Carbon");
+  if (!ident_id.is_valid()) {
+    return SemIR::NameScopeId::Invalid;
+  }
+  auto name_id = SemIR::NameId::ForIdentifier(ident_id);
+
+  // Look up `package.Carbon`.
+  auto package_id = context.LookupQualifiedName(
+      node_id, name_id, SemIR::NameScopeId::Package, /*required=*/false);
+  if (!package_id.is_valid()) {
+    return SemIR::NameScopeId::Invalid;
+  }
+
+  package_id = context.constant_values().Get(package_id).inst_id();
```

Comment:
Why does this re-resolve through the constant? (a comment might help)

Reply:
Comment added.